runners/spark/src/main/java/org/apache/beam/runners/spark/io/SourceRDD.java

```diff
@@ -119,21 +133,25 @@ public Bounded(SparkContext sc,
 
         @Override
         public boolean hasNext() {
-          try {
-            if (!started) {
-              started = true;
-              finished = !reader.start();
-            } else {
-              finished = !reader.advance();
-            }
-            if (finished) {
-              // safely close the reader if there are no more elements left to read.
+          try (Closeable ignored = MetricsEnvironment.scopedMetricsContainer(metricsContainer)) {
```

Comment:
It would be great to have a comment here, stating that this scoping is needed here so that metrics in other places (e.g. `UnboundedReader#advance()`) work.